.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,34 @@
+# This workflow will run tests using node and then publish a package to npm when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - run: pnpm install
+      - run: pnpm build
+      - run: pnpm test
+
+  publish-npm:
```

Comment:
You can do this in a single job, since everything is dependent on what came before.  I also think you have to change how you do the `NPM_TOKEN`.  @tarasglek you need to add a secret, `NPM_TOKEN`, scoped to your repo.  Then you'll own this.

```yaml
  build:
    runs-on: ubuntu-latest
    env:
      NPM_TOKEN: ${{secrets.NPM_TOKEN}}
    steps:
      - uses: actions/checkout@v3
      - name: Set NPM_TOKEN in .npmrc
        run: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
      - uses: pnpm/action-setup@v2
        with:
          version: 8
          run_install: true
      - run: pnpm build
      - run: pnpm test
      - run: pnpm publish
```